src/dali/plugin/input/rest/kucoin_futures.py

```diff
@@ -0,0 +1,280 @@
+# Copyright 2022 topcoderasdf
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Kucoin Futures REST plugin links:
+# REST API: https://docs.kucoin.com/futures/#general
+# Authentication: https://docs.kucoin.com/futures/#authentication
+# Endpoint: https://api-futures.kucoin.com
+
+import json
+import time
+from datetime import datetime
+from decimal import Decimal
```

Comment:
Use RP2Decimal instead of Decimal: it has custom high-precision math that it used throughout the code base.

Reply:
fixed